gulp-tasks/clean.js

```diff
@@ -1,33 +1,22 @@
-'use strict';
+/**
+ * Clean tasks - Clean compiled dist directories.
+ */
 
 // Include Our Plugins
 const del = require('del');
+const cFlags = require("./utils/cflags");
+const dutil = require("./utils/doc-util");
 
-// Export our tasks.
-module.exports = {
-
-  // Clean CSS files.
-  cleanCSS: function() {
-    return del(['./dist/css/*'], { force: true });
-  },
-
-  // Clean Fonts files.
-  cleanFonts: function() {
-    return del(['./dist/fonts/*'], { force: true });
-  },
-
-  // Clean Images files.
-  cleanImages: function() {
-    return del(['./dist/img/*'], { force: true });
-  },
-
-  // Clean JS files.
-  cleanJS: function() {
-    return del(['./dist/js/*'], { force: true });
-  },
-
-   // Clean Sass files.
-   cleanSass: function() {
-    return del(['./dist/scss/*'], { force: true });
+// Clean generated Dist directory.
+exports.cleanDist = done => {
+  if (!cFlags.cleanup) {
+    dutil.logMessage(
+      "clean-dist",
+      "Skipping cleaning up the distribution directories."
+    );
+    return done();
   }
+  dutil.logMessage("clean-dist", "Removing distribution directories.");
+
+  return del("dist");
 };
```

Comment:
The previous `cleanCSS, cleanFonts, cleanImages, cleanJS` weren't used anywhere. The only real clean task existed elsewhere; which I moved here.